clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/DistributableWebResourceDefinition.java

```diff
@@ -0,0 +1,137 @@
+        private final org.jboss.as.clustering.controller.Capability capability;
+
+        Capability(Requirement requirement) {
+            this.capability = new RequirementCapability(requirement);
+        }
+
+        @Override
+        public org.jboss.as.clustering.controller.Capability getCapability() {
+            return this.capability;
+        }
+    }
+
+    enum Attribute implements org.jboss.as.clustering.controller.Attribute {
+        DEFAULT_SESSION_MANAGEMENT("default-session-management", ModelType.STRING, new CapabilityReference(Capability.DEFAULT_SESSION_MANAGEMENT_PROVIDER, WebProviderRequirement.SESSION_MANAGEMENT_PROVIDER)),
+        DEFAULT_SSO_MANAGEMENT("default-single-sign-on-management", ModelType.STRING, new CapabilityReference(Capability.DEFAULT_SSO_MANAGEMENT_PROVIDER, WebProviderRequirement.SSO_MANAGEMENT_PROVIDER)),
+        ;
+        private final AttributeDefinition definition;
+
+        Attribute(String name, ModelType type, CapabilityReferenceRecorder reference) {
+            this.definition = new SimpleAttributeDefinitionBuilder(name, type)
+                    .setAllowExpression(false)
+                    .setRequired(true)
+                    .setCapabilityReference(reference)
+                    .setFlags(Flag.RESTART_RESOURCE_SERVICES)
+                    .build();
+        }
+
+        @Override
+        public AttributeDefinition getDefinition() {
+            return this.definition;
+        }
+    }
+
+    DistributableWebResourceDefinition() {
+        super(PATH, DistributableWebExtension.SUBSYSTEM_RESOLVER);
+    }
+
+    @Override
+    public void register(SubsystemRegistration parent) {
+        ManagementResourceRegistration registration = parent.registerSubsystemModel(this);
+
+        new DefaultSubsystemDescribeHandler().register(registration);
+
+        ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
+                .addAttributes(Attribute.class)
+                .addCapabilities(Capability.class)
+                .addRequiredSingletonChildren(LocalRoutingProviderResourceDefinition.PATH)
+                ;
+        ResourceServiceHandler handler = new DistributableWebResourceServiceHandler();
+        new DeploymentChainContributingResourceRegistration(descriptor, handler, this).register(registration);
+
+        new LocalRoutingProviderResourceDefinition().register(registration);
+        new InfinispanRoutingProviderResourceDefinition().register(registration);
+
+        new InfinispanSessionManagementResourceDefinition().register(registration);
+        new InfinispanSSOManagementResourceDefinition().register(registration);
+    }
+
+    @Override
+    public void accept(DeploymentProcessorTarget target) {
+        JBossAllXmlParserRegisteringProcessor.Builder builder = JBossAllXmlParserRegisteringProcessor.builder();
+        for (DistributableWebDeploymentSchema schema : EnumSet.allOf(DistributableWebDeploymentSchema.class)) {
```

Comment:
This could just use `DistributableWebDeploymentSchema.values()`.

Reply:
EnumSet.allOf(...) iteration is more efficient than Enum.values().